### DISCUSSION TITLE
configuring a plugin in webpack.

### DISCUSSION THREAD

**Discussion by kalingaCoder** [nrwl/nx#15932] [General]
Subj: Need help in configuring a plugin in webpack
My webpack configuration:
-------------------------------

```js
const { withReact } = require("@nrwl/react");
const BrotliPlugin = require("brotli-webpack-plugin");
const { composePlugins, withNx } = require("@nrwl/webpack");
const { merge } = require("webpack-merge");
const withModuleFederation = require("@nrwl/react/module-federation");
const moduleFederationConfig = {
  name: "cart",
  exposes: {
    "./Module": "./src/remote-entry.ts",
  },
};
 const brotliInstance = ()=> new BrotliPlugin({
   asset: "[path].br",
   threshold: 0,
   minRatio: 0.8,
 });
   (module.exports = composePlugins(
     withNx(),
     async (config, { options, context }) => {
       const federatedModules = await withModuleFederation({
         ...moduleFederationConfig,
       });
       console.info({
         config,
         context,
         federatedModules: federatedModules(config, context),
       });
       config.plugins.push(brotliInstance());
       return merge(federatedModules(config, context), {
         // overwrite values here
       });
     }
   ));
```

   -------------------------------
Error I  get when building():
     NX   Cannot read properties of undefined (reading 'length')
     
My goal is to configure moduleFederationConfig, brotli plugin.

**Steps to replicate:**
1. Repo URL: https:[//stackblitz.com/edit/react-ts-9b3wcs?file=apps/cart/webpack.config.js] (https://nrwlcommunity.slack.com//stackblitz.com/edit/react-ts-9b3wcs?file=apps/cart/webpack.config.js) or https://github.com/kalingaCoder/react_MF
2. npm run build-cart (edited)

**Comment by Coly010** [comment]
There is now some information on the docs here: https://nx.dev/recipes/module-federation/faster-builds#what-was-generated
